### PR TITLE
fix salt-api's default opts were covered by salt-master

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
### What does this PR do?

salt-api default pidfile : salt-api.pid ; log_file: log/api were covered by salt-master's salt-master.pid and log/master
### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by rallytime** [comment]
Hi @xiaoanyunfei - Looks like there is a pylint error here. Mind cleaning that up? https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/4540/violations/file/salt/config/__init__.py/

@whiteinge Can you review this when you get a moment?

**Comment by whiteinge** [comment]
I need to defer to @s0undt3ch's expertise for this review.

File: salt/config/__init__.py [hunk 1]
```diff
@@ -3273,10 +3273,11 @@ def api_config(path):
     '''
     # Let's grab a copy of salt's master default opts
     defaults = DEFAULT_MASTER_OPTS
+    opts = client_config(path, defaults=defaults)
     # Let's override them with salt-api's required defaults
-    defaults.update(DEFAULT_API_OPTS)
+    opts.update(DEFAULT_API_OPTS)
```
Comment (s0undt3ch) [inline salt/config/__init__.py]: Applying the defaults against loaded opts is not good. You override configured settings with the defaults.

**Comment by s0undt3ch** [comment]
Using syspaths is what should be done, I just don't understand why there was a need to change the rest.

**Comment by s0undt3ch** [comment]
And we should cover this with a unit test..

File: salt/config/__init__.py (hunk 1 above)
Comment (xiaoanyunfei) [inline salt/config/__init__.py]: @s0undt3ch If I set the "pidfile" and "log_file" in /etc/salt/master, the salt-api will use the same pid and log_file as salt-master too. Their logs  were mixed up and pidfile were covered by each other. Using the default may not be the best solution. Maybe we should use api_log_file and api_pidfile, just like syndic_log_file and syndic_pidfile

**Comment by xiaoanyunfei** [comment]
@s0undt3ch If I set the "pidfile" and "log_file" in /etc/salt/master, the salt-api will use the same pid and log_file as salt-master too. Their logs were mixed up and pidfile were covered by each other. Using the default may not be the best solution. Maybe we should use api_log_file and api_pidfile, just like syndic_log_file and syndic_pidfile

**Comment by s0undt3ch** [comment]
@xiaoanyunfei yes, having the same approach we have for syndic applied to the salt-api is the reasonable approach to follow.

**Comment by cachedout** [comment]
Hi @xiaoanyunfei. What's the status on this one? Are you planning to make additional changes?

**Comment by xiaoanyunfei** [comment]
@cachedout I will make some additional changes before this weekend

**Pull request closed without merging**